thunderstore-components/src/components/PackageUpload.tsx

```diff
@@ -1,7 +1,142 @@
-import React from "react";
+import {
+  Box,
+  Button,
+  Center,
+  CloseButton,
+  Heading,
+  Select,
+  Slide,
+  Text,
+  useColorModeValue,
+  useToast,
+} from "@chakra-ui/react";
+import JSZip from "jszip";
+import React, { useEffect, useState } from "react";
+import { FileUpload } from "./FileUpload";
+import { Markdown } from "./Markdown";
+import { StickyFooter } from "./StickyFooter";
 
-interface PackageUploadProps {}
+export const PackageUpload: React.FC<never> = () => {
```

Comment:
This component seems quite massive, and I'd recommend splitting it into smaller pieces if you can.

Reply:
I agree, I'm not too sure where to split it though, at least in a way that would be easily reusable